packages/core/src/actions.ts

```diff
@@ -609,7 +609,7 @@ export function choose<TContext, TEvent extends EventObject>(
 
 export function resolveActions<TContext, TEvent extends EventObject>(
   machine: StateNode<TContext, any, TEvent, any, any, any>,
-  currentState: State<TContext, TEvent> | undefined,
+  currentState: State<TContext, TEvent, any, any, any> | undefined,
```

Comment:
Can use `AnyState` here, no? (nit)

Reply:
This would have a slightly different meaning though as we wouldn't require `State<...>` to be based on `TContext` and `TEvent`. I think it would be somewhat OK to change this *here* as `resolveActions` is really just a private helper but we couldn't make such a change in the whole codebase.

EIther way - I think we should leave such refactorings for the next branch after we freeze v4 development.